DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -389,7 +438,225 @@ void ZCounting::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+
+  // Get SuperClusters
+  edm::Handle<edm::View<reco::SuperCluster> > superclusters;
+  iEvent.getByToken(fSCName_token, superclusters);
+
+  // Get Rho
+  edm::Handle<double> rhoHandle;
+  iEvent.getByToken(fRhoToken, rhoHandle);
+  EleID_.setRho(*rhoHandle);
+
+  // Get beamspot
+  edm::Handle<reco::BeamSpot> beamspotHandle;
+  iEvent.getByToken(fBeamspotToken, beamspotHandle);
+  EleID_.setBeamspot(beamspotHandle);
+
+  // Conversions
+  edm::Handle<reco::ConversionCollection> conversionsHandle;
+  iEvent.getByToken(fConversionToken, conversionsHandle);
+  EleID_.setConversions(conversionsHandle);
+
+  TLorentzVector vTag(0.,0.,0.,0.);
+  TLorentzVector vProbe(0.,0.,0.,0.);
+  TLorentzVector vDilep(0.,0.,0.,0.);
+  edm::Ptr<reco::GsfElectron> eleProbe;
+  enum { eEleEle2HLT=1, eEleEle1HLT1L1, eEleEle1HLT, eEleEleNoSel, eEleSC };  // event category enum
+
+  // Loop over Tags
+  for (size_t itag = 0; itag < electrons->size(); ++itag){
+    const auto el1 = electrons->ptrAt(itag);
+    if( not EleID_.passID(el1) ) continue;
+
+    float pt1  = el1->pt();
+    float eta1 = el1->eta();
+    float phi1 = el1->phi();
+
+    if(!isElectronTriggerObj(*fTrigger, TriggerTools::matchHLT(eta1, phi1, fTrigger->fRecords, *hTrgEvt))) continue;
+    vTag.SetPtEtaPhiM(pt1, eta1, phi1, ELECTRON_MASS);
+
+
+    // Tag selection: kinematic cuts, lepton selection and trigger matching
+    double tag_pt = vTag.Pt();
+    double tag_abseta = fabs(vTag.Eta());
+
+    bool tag_is_valid_tag = ele_tag_selection(tag_pt,tag_abseta);
+    bool tag_is_valid_probe = ele_probe_selection(tag_pt,tag_abseta);
+
+    if( not (tag_is_valid_tag or tag_is_valid_probe) ) continue;
+
+    // Loop over probes
+    for (size_t iprobe = 0; iprobe < superclusters->size(); ++iprobe){
+      // Initialize probe
+      const auto sc = superclusters->ptrAt(iprobe);
+      if(*sc == *(el1->superCluster())) {
+        continue;
+      }
+
+      // Find matching electron
+      for (size_t iele = 0; iele < electrons->size(); ++iele){
+        if(iele == itag) continue;
+        const auto ele = electrons->ptrAt(iele);
+        if(*sc == *(ele->superCluster())) {
+          eleProbe = ele;
+          break;
+        }
+      }
+
+      // Assign final probe 4-vector
+      if(eleProbe.isNonnull()){
+        vProbe.SetPtEtaPhiM( eleProbe->pt(), eleProbe->eta(), eleProbe->phi(), ELECTRON_MASS);
+      } else {
+        double pt = sc->energy() * TMath::Sqrt( 1 - pow(TMath::TanH(sc->eta()),2) );
+        vProbe.SetPtEtaPhiM( pt, sc->eta(), sc->phi(), ELECTRON_MASS);
+      }
+
+      // Probe Selection
+      double probe_pt = vProbe.Pt();
+      double probe_abseta = fabs(sc->eta());
+      bool probe_is_valid_probe = ele_probe_selection(probe_pt, probe_abseta);
+      if( !probe_is_valid_probe ) continue;
+
+      // Good Probe found!
+
+      // Require good Z
+      vDilep = vTag + vProbe;
+      float MASS_LOW = 80.0;
+      float MASS_HIGH = 100.0;
```

Comment:
perhaps these numbers should be made part of the configuration? If not, please document them.

Reply:
Done.